### PR TITLE
[Fix] MMEditing cannot save results when testing

### PR DESCRIPTION
**Pull request by SingleZombie**
## Motivation

We can save results when using `tools/test.py`. However, MMEditing fails to do this because we lack some codes.

## Modification

Copy the codes related to save results from `mmedit.basic_restorer.py`  to `mmdeploy.super_resolution_model.py`

### CODE REVIEW

File: mmdeploy/codebase/mmedit/deploy/super_resolution_model.py [hunk 1]
```diff
@@ -88,6 +89,9 @@ def forward(self,
     def forward_test(self,
                      lq: torch.Tensor,
                      gt: Optional[torch.Tensor] = None,
+                     meta=None,
```
Comment (grimoire) [inline mmdeploy/codebase/mmedit/deploy/super_resolution_model.py:92]: Please add type hint.

File: mmdeploy/codebase/mmedit/deploy/super_resolution_model.py [hunk 2]
```diff
@@ -96,6 +100,8 @@ def forward_test(self,
             lq (torch.Tensor): The input low-quality image of the model.
             gt (torch.Tensor): The ground truth of input image. Defaults to
                 `None`.
+            save_image (bool): Whether to save image. Default: False.
```
Comment (grimoire) [inline mmdeploy/codebase/mmedit/deploy/super_resolution_model.py:103]: Where is the root docstring of `meta`?

File: mmdeploy/codebase/mmedit/deploy/super_resolution_model.py [hunk 3]
```diff
@@ -104,6 +110,17 @@ def forward_test(self,
         """
         outputs = self.forward_dummy(lq)
         result = self.test_post_process(outputs, lq, gt)
+
+        # Align to mmediting BasicRestorer
+        if save_image and save_path:
```
Comment (AllentDan) [inline mmdeploy/codebase/mmedit/deploy/super_resolution_model.py:115]: seems `save_image` and `save_path` both need to be specified if I want to save. Keeping only `save_path` is enough then.

Comment (SingleZombie) [reply mmdeploy/codebase/mmedit/deploy/super_resolution_model.py:115]: I agree. Our `tools/test.py` uses `--show` and `--show-dir` to control the output behavior. Maybe it should also be simplified later.

**Review by AllentDan — APPROVED** [review]
LGTM after adding type hint.

**Review by grimoire — APPROVED** [review]

**Pull request merged**